src/link/script.y

```diff
@@ -251,10 +251,16 @@ yy::parser::symbol_type yylex() {
 	if (c == EOF) {
 		// Basically yywrap().
 		if (lexerStack.size() != 1) {
-			lexerStack.pop_back();
-			return yylex();
+			if (!atEof) {
+				// Inject a newline at EOF to simplify parsing.
```

Comment:
It would be much cleaner to have a separate `line` production and treat `\n` as a line separator rather than terminator. Can this be reasonably done?

Reply:
Then let's take the hack.